Add recent games tab

Luchanso opened the pull request:
* Fixed warnings about prop-types (my console was red before it)
* Add ErrorBox component
* Add tab "Recent" on hero page
![image](https://user-images.githubusercontent.com/2098777/31354576-88cdbd02-ad3f-11e7-8c54-ee106b6bde96.png)

@howardchung, your https://github.com/odota/core/pull/1542 branch not work on my local machine. I switched branch, added Steam API Key and wait data from Dota API, next I try fetch data "http://localhost:5000/api/heroes/1/matches?", and got empty array in response

Same problem with '/api/proMatches' and '/api/publicMatches?mmr_descending=1' endpoints

howardchung commented:
This only counts pro matches, so you probably just don't have one of those.  You can look up a pro match with antimage via the Explorer and then request a parse of that match on your local machine

howardchung commented:
Suggestions:
* Add k/d/a for each match (might need to update the query in core)
* Make an "overview" section similar to what we have for players, where we request multiple endpoints and display multiple tables

Luchanso commented:
@howardchung yes, I don't have pro matches on my local michine
Okey, I will add k/d/a and overview :ok_hand:

Luchanso commented:
* Added pagination
* Added K/D/A
![image](https://user-images.githubusercontent.com/2098777/31380675-ac3f7ea8-adba-11e7-8e07-4b45353420d9.png)

Now I doing "overviews", and maybe will do other things from #1143

Luchanso commented:
* Add hero attributes
* Add hero roles
![image](https://user-images.githubusercontent.com/2098777/31387265-7528f54e-add2-11e7-88f9-af05b26bdfbe.png)

Luchanso commented:
I need some time for thinking about other data:

"icon"
"base_health"
"base_health_regen"
"base_mana"
"base_mana_regen"
"base_armor"
"base_mr"
"base_attack_min"
"base_attack_max"
"attack_range"
"projectile_speed"
"attack_rate"
"move_speed"
"turn_rate"
"cm_enabled"
"legs" 😄 👣
"pro_ban"
"hero_id"
"pro_win"
"pro_pick"
"2000_pick"
"2000_win"
"5000_pick"
"5000_win"
"1000_pick"
"1000_win"
"3000_pick"
"3000_win"
"4000_pick"
"4000_win"

Luchanso commented:
In progress

Luchanso commented:
* [x] Add other hero attributes
![image](https://user-images.githubusercontent.com/2098777/31495129-0120f3ca-af5f-11e7-9bd8-e0d66180800e.png)
And now:
* [ ] I changing text on icons
* [ ] Fixing some bugs with adoptive layout

Luchanso commented:
Change style and colors

![image](https://user-images.githubusercontent.com/2098777/31496903-dd71b87c-af65-11e7-83be-74f7d6c58b80.png)

* [ ] Fixing some bugs with layout on mobile

Luchanso commented:
I found bug and fixed it. Bug about languages key:

![image](https://user-images.githubusercontent.com/2098777/31498671-44e480d4-af6b-11e7-8d84-a4743d4cd8e5.png)


Fixed version:
If key not found in selected language, then we take this key from en_US language
![image](https://user-images.githubusercontent.com/2098777/31498692-56942190-af6b-11e7-97f7-93cb4f95a7dd.png)

@howardchung and I'm done. Result:
![image](https://user-images.githubusercontent.com/2098777/31498758-8a9d6212-af6b-11e7-8759-222af41e155e.png)

Luchanso commented:
I will add some ui for duration/players/matchups in this PR later, okey?

howardchung commented:
The match list should probably include the player who played the hero in that game.

Luchanso commented:
Okey, I will add it

Luchanso commented:
* [x] Added column account id for recent matches
![image](https://user-images.githubusercontent.com/2098777/31575025-78dadd9a-b0e5-11e7-977f-2400746eb183.png)

Luchanso commented:
I'm added matchups page
![image](https://user-images.githubusercontent.com/2098777/31575799-2c56ed12-b0f7-11e7-8e60-c6d170b60383.png)

It remains to add the /heroes/{heroId}/durations and /heroes/{heroId}/players

howardchung commented:
For durations you can use the HistogramGraph visualization that's already implemented.

We probably want to add a note somewhere that this data is from pro matches only.

Maybe we should retire the benchmarks section since that's from public games.

Luchanso commented:
Okey, I'll add note "* All data are taken from PRO games"

I think we just need add note in benchmarks tab "* Benchmarks data are taken from Public games"

And I don't understand "durations", can you explain me, what is it?

howardchung commented:
It's the win rate of the hero for each range of 5 minutes (rounded down).

0-5 minutes, 5-10 minutes, 10-15 minutes, etc.  Basically a measure of
whether a hero is strong late game or early game.

The histogram has a bar indicating the number of games in each bucket with
a color indicating the win rate (greener = higher win %)

On Sat, Oct 14, 2017 at 12:47 PM, Loutchansky Oleg <notifications@github.com
> wrote:

> Okey, I'll add note "* All data are taken from PRO games"
>
> I think we just need add note in benchmarks tab "* Benchmarks data are
> taken from Public games"
>
> And I don't understand "durations", can you explain me, what is it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/odota/ui/pull/1247#issuecomment-336662311>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AC_UODbQ0gjzTcYYkMvapA-agQKKaoRUks5ssQ_pgaJpZM4Py8Yl>
> .
>

Luchanso commented:
Thank you, now I unserstand what is it.

blukai commented:
@Luchanso maybe make hero page look similar to how github profile looks like?

shitty photoshop: 🦄 
![- -1](https://user-images.githubusercontent.com/20866892/31673137-1c93e368-b367-11e7-9864-4ee1153deed6.jpg)

Luchanso commented:
@blukai Okey, I will experiment with it

Luchanso commented:
Finished tab "Durations"
![image](https://user-images.githubusercontent.com/2098777/31675026-2818b1be-b36c-11e7-80e5-ad8a414dec74.png)

Now I doing /players page and experement with layout

Luchanso commented:
Done /players page
![image](https://user-images.githubusercontent.com/2098777/31676840-867b7296-b371-11e7-9e42-309ff779eb13.png)

I will resolve conflicts and try to change layout

Luchanso commented:
@blukai, @howardchung, I was changed layout on heroes page. If you like this, you can review and merge, if not, I can remove changes or I'm ready to listen to your suggestions

![image](https://user-images.githubusercontent.com/2098777/31773137-71779df0-b4ea-11e7-9a51-63690608d146.png)

Luchanso commented:
Ahh, want remove useful (duplicates) labels in tab

Luchanso commented:
Done

howardchung commented on src/components/Hero/AttributesBlock.jsx:
```diff
@@ -0,0 +1,134 @@
+
+const renderIcon = Icon => <Icon style={{ width: 20, marginRight: 4 }} />;
+
+const calcArmorPercent = hero => Math.round(0.06 * hero / (1 + (0.06 * hero)) * 100);
```
I'm not quite sure what this is, can you annotate/comment it?

Luchanso replied:
Yes, sure

howardchung commented:
Comments:

* I think each tab should have the Heading component, and the subtitle value should be "Data from public matches" or "Data from professional matches"
* Recent matches tab should have the player's name displayed, not their account ID
* KDA isn't showing up (need to pass the data down from core?)
* How is the advantage column calculated in the Matchups tab?
* Matchups should be sorted by number of games descending (can probably be done on backend)
* Players data is from pro games, not public
* Same issue with showing the account ID instead of the name
* Players list should be sorted in descending order (can probably be done by changing the backend query)

Luchanso commented:
Okay, I'll do

Luchanso commented:
Tasklist:
- [x] I think each tab should have the Heading component, and the subtitle value should be "Data from public matches" or "Data from professional matches"
- [x] Recent matches tab should have the player's name displayed, not their account ID
- [ ] KDA isn't showing up (need to pass the data down from core?)
- [ ] How is the advantage column calculated in the Matchups tab?
- [ ] Matchups should be sorted by number of games descending (can probably be done on backend)
- [x] Players data is from pro games, not public
- [ ] Same issue with showing the account ID instead of the name
- [ ] Players list should be sorted in descending order (can probably be done by changing the backend query)

Luchanso commented:
@howardchung anything else?

howardchung commented:
Some final tweaks:
* Use the small muted font for event name on https://deploy-preview-1247--chef-sebastian-43878.netlify.com/heroes/102/recent (see /matches page for example)
* https://deploy-preview-1247--chef-sebastian-43878.netlify.com/heroes/102/matchups it's still not clear to the end user how the advantage column is calculated.  I assume this is the Wilson score?  We might want to add a tooltip explaining what this is
* https://deploy-preview-1247--chef-sebastian-43878.netlify.com/heroes/102/players use 2 decimal places for win % and remove the % (consistency with other tab)

Luchanso commented:
* [x] Use the small muted font for event name on /heroes/{heroId}/recent
* [x] Tooltip for advantage column about Wilson score
* [x] Use 2 decimal places for win % and remove the % (consistency with other tab)

@howardchung, can you review again?

howardchung commented:
LGTM, thanks for the contribution!

Luchanso commented:
@howardchung thank you too, it was good experience. I'll try continue contribute in free time :+1:

howardchung commented:
If you want to work on something similar teams pages are still available #1142
